src/writer/command_line.py

```diff
@@ -1,133 +1,78 @@
-        writer.serve.serve(
-            absolute_app_path, mode="run", port=port, host=host, enable_server_setup=True)
-    elif command in ("hello"):
-        create_app("hello", template_name="hello", overwrite=True)
-        writer.serve.serve("hello", mode="edit",
-                               port=port, host=host, enable_remote_edit=enable_remote_edit,
-                               enable_server_setup=False)
-    elif command in ("create"):
-        create_app(absolute_app_path, template_name=template_name)
+    pass
+
+@main.command()
+@click.option('--host', default="127.0.0.1", help="Host to run the app on")
+@click.option('--port', default=5000, help="Port to run the app on")
+@click.argument('path')
+def run(path, host, port):
+    """Run the app from PATH folder in run mode."""
+
+    abs_path, is_folder = _get_absolute_app_path(path)
+    if not is_folder:
+        raise click.ClickException("A path to a folder containing a Writer Framework app is required. For example: writer run my_app")
+
+    writer.serve.serve(
+        abs_path, mode="run", port=port, host=host, enable_server_setup=True)
+
+@main.command()
+@click.option('--host', default="127.0.0.1", help="Host to run the app on")
+@click.option('--port', default=5000, help="Port to run the app on")
+@click.option('--enable-remote-edit', help="Set this flag to allow non-local requests in edit mode.", is_flag=True)
+@click.option('--enable-server-setup', help="Set this flag to enable server setup hook in edit mode.", is_flag=True)
+@click.argument('path')
+def edit(path, port, host, enable_remote_edit, enable_server_setup):
+    """Run the app from PATH folder in edit mode."""
+
+    abs_path, is_folder = _get_absolute_app_path(path)
+    if not is_folder:
+        raise click.ClickException("A path to a folder containing a Writer Framework app is required. For example: writer edit my_app")
+
+    writer.serve.serve(
+        abs_path, mode="edit", port=port, host=host,
+        enable_remote_edit=enable_remote_edit, enable_server_setup=enable_server_setup)
+
+@main.command()
+@click.argument('path', type=click.Path(exists=True, file_okay=False, dir_okay=True, resolve_path=True))
+@click.option('--template', help="The template to use when creating a new app.")
+def create(path, template):
+    """Create a new app in PATH folder."""
+
+    abs_path, _ = _get_absolute_app_path(path)
+    if abs_path is None:
+        raise click.ClickException("A target folder is required to create a Writer Framework app. For example: writer create my_app")
+
+    create_app(abs_path, template_name=template)
+
+@main.command()
+@click.option('--host', default="127.0.0.1", help="Host to run the app on")
+@click.option('--port', default=5000, help="Port to run the app on")
+@click.option('--enable-remote-edit', help="Set this flag to allow non-local requests in edit mode.", is_flag=True)
+def hello(port, host, enable_remote_edit):
+    """Create and run a onboarding 'Hello' app."""
```

Comment:
*an onboarding

Are these messages shown in the CLI? with --help?

Reply:
Fixed and yes